src/opm/output/eclipse/AggregateWellData.cpp

```diff
@@ -661,9 +668,30 @@ namespace {
             // Not fully characterised.
             xWell[Ix::item38] = xWell[Ix::GasPrRate];
 
+            xWell[Ix::PrimGuideRate] = xWell[Ix::PrimGuideRate_2] = -get("WGIGR");
+
             xWell[Ix::GasVoidPrRate] = xWell[Ix::VoidPrRate];
         }
 
+        template <class XWellArray>
+        void assignOilInjector(const std::string&         well,
+                               const ::Opm::SummaryState& smry,
+                               XWellArray&                xWell)
+        {
+            using Ix = ::Opm::RestartIO::Helpers::VectorItems::XWell::index;
+
+            auto get = [&smry, &well](const std::string& vector)
+            {
+                const auto key = vector + ':' + well;
+
+                return smry.has(key) ? smry.get(key) : 0.0;
```

Comment:
Here you can use ` SummaryState::has_well_var(well, var)`

Reply:
Yes...kind of.  I thought that `SummaryState::has_well_var()` required that the value was stored using `SummaryState::update_well_var()` and I'm not sure if we can guarantee that.  If there is no such requirement then I'll happily switch to using `has_well_var()`.

Reply:
You are correct that `SummaryState::has_well_var()` requires that the value has been added using `::update_well_var()` - but that requirement is already hard wired; and e.g. `UDQ / ACTIONX` will fail miserably if that is not the case.

Could be better documented / tested though; could of course also consider splitting up the composite key in the `::update( )` - that way we could *guarantee* that the `has_well_var()` and `has_group_var()` can be safely used.